### PR TITLE
Restructure examples for TF, and removed example notebooks everywhere (except 1)

### PR DESCRIPTION
**Pull request by rahul003**
### Description of changes:
Arranging examples in the following structure
- examples
       - local
       - sagemaker_byoc
       - sagemaker_official_container

Haven't removed the tensor analysis notebook for mxnet yet as it is not yet converted to SM notebook

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rahul003** [comment]
Addressing your comments

**Comment by rahul003** [comment]
Moved scripts out, and fixed couple of issues in the resnet script

**Comment by rahul003** [comment]
Added main guards to local scripts, but don't want to change the sagemaker scripts just yet, as they are in the notebooks in this form. We can change them later.

Removed unsupported distributed training examples for now, can bring them back once supported. Also removed the mirrored strategy script as it's crashing even without the Hook

File: docs/sagemaker.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Here's a list of frameworks and versions which support this experience.
 | [TensorFlow](tensorflow.md) | 1.15 |
 | [MXNet](mxnet.md) | 1.6 |
 | [PyTorch](pytorch.md) | 1.3 |
-| [XGBoost](xgboost.md) | |
+| [XGBoost](xgboost.md) | >=0.90-2 [As Built-in algorithm](xgboost.md#use-xgboost-as-a-built-in-algorithm)|
```
Comment (jarednielsen) [inline docs/sagemaker.md]: Will this be >= or just == ?

File: README.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - [Overview](#overview)
 - [Examples](#sagemaker-example)
```
Comment (jarednielsen) [inline README.md]: This link is broken

**Comment by jarednielsen** [comment]
Lgtm if CI passes.

File: docs/sagemaker.md (hunk 1 above)
Comment (rahul003) [reply docs/sagemaker.md]: from xgboost doc `version `0.90-2` or later`

File: README.md (hunk 2 above)
Comment (rahul003) [reply README.md]: fixed thanks

**Pull request merged**